javascript/node/index.js

```diff
@@ -34,11 +34,14 @@ const main = async () => {
             typeRegistry: registry,
         },
     });
+
+    let streaming = true;
     
     // The infite loop handles disconnections. Every time an disconnection error is thrown, the loop will automatically reconnect
     // and start consuming from the latest commited cursor.
-    while (true) {
+    while (streaming) {
         try {
+            streaming = false;
```

Comment:
Why this was refactored like this, did you hit some problem? I find it personally harder to reason it about in the new way. It feels weird to me to set `streaming = false` right before calling `await stream`, since if we call stream, we are effectively streaming no?